extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/ApplicationCoroutineScope.kt

```diff
@@ -0,0 +1,46 @@
+package org.jboss.resteasy.reactive.server.runtime.kotlin
+
+import io.vertx.core.Context
+import kotlinx.coroutines.*
+import org.eclipse.microprofile.context.ManagedExecutor
+import javax.annotation.PreDestroy
+import javax.enterprise.context.ApplicationScoped
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Application wide coroutine scope. Should start and die with the rest of the application, along with child coroutines
+ * (structured concurrency).
+ *
+ * This should be the main interception point to place user specific [CoroutineContext.Element] objects. Things
+ * like activity ids, logger context propagation, tracing, exception handlers, etc.
+ */
+@ApplicationScoped
+class ApplicationCoroutineScope : CoroutineScope, AutoCloseable {
+    override val coroutineContext: CoroutineContext = SupervisorJob()
+
+    @PreDestroy
+    override fun close() {
+        coroutineContext.cancel()
+    }
+}
+
+/**
+ * Dispatches the coroutine in a worker thread from Vertx.
+ */
+class VertxDispatcher(private val vertxContext: Context) : CoroutineDispatcher() {
+    override fun dispatch(context: CoroutineContext, block: Runnable) {
+        vertxContext.runOnContext {
+            block.run()
```

Comment:
This will not have context propagation: you need to use `ThreadContext` and capture context at the right place to restore it around `block.run()`.

Reply:
Thanks for the advice. Will use it based on the response to the earlier comment about context propagation.

Reply:
We also need to initialize/propagate the request context when we start the dispatch. Currently the request context is not activated for suspended functions.

Reply:
This does not have to happen in this PR though, as it is a bit of a different issue, but we can't really fix it until this is in as it would conflict.

Reply:
Agreed that enabling that would be out of scope for this PR.
Will leave a comment stating that context propagation will need to be handled later.